praw/models/util.py

```diff
@@ -166,11 +169,12 @@ def stream_generator(function, pause_after=None, skip_existing=False):
             without_before_counter = (without_before_counter + 1) % 30
         for item in reversed(list(function(
                 limit=limit, params={'before': before_fullname}))):
-            if item.fullname in seen_fullnames:
+            key = getattr(item, key_name)
+            if key in seen_fullnames:
```

Comment:
If `key` is renamed `attribute` then it might make sense to rename `seen_fullnames` to `seen_attributes`.